cli/src/main/assembly/exec.xml

```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<assembly xmlns="http://maven.apache.org/plugins/maven-assembly-plugin/assembly/1.1.2" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/plugins/maven-assembly-plugin/assembly/1.1.2 http://maven.apache.org/xsd/assembly-1.1.2.xsd">
+  <id>${os.detected.classifier}</id>
+  <formats>
+    <format>tar.gz</format>
+  </formats>
+  <includeBaseDirectory>false</includeBaseDirectory>
+  <fileSets>
+    <fileSet>
+      <directory>${project.build.directory}</directory>
+      <outputDirectory>./</outputDirectory>
+      <excludes>
+        <exclude>*</exclude>
+        <exclude>*/**</exclude>
+      </excludes>
+    </fileSet>
+    <fileSet>
+      <directory>${project.build.directory}</directory>
+      <outputDirectory>./</outputDirectory>
+      <includes>
+        <include>${imageName}-windows.exe</include>
+        <include>${imageName}-linux</include>
+        <include>${imageName}-mac</include>
```

Comment:
I am not sure, if I get this correctly:
Are we building one generic *.tar.gz that contains the binaries for all operating systems?
Or do we build a different *.tar.gz release artifact per OS?
I thought we would do the latter and maybe we do.

Some further thoughts for future improvements if we do the latter:
* The binary inside the package should be named just `bin/ide[.exe]`.
* The download file of the release should be rather `ide-cli-«os»-«arch».tar.gz` where `arch` is most probably `x64` but for `mac` we could think of providing `arm64`.

Reply:
Currently we build three different `*.tar.gz `. 

We can also create one big `tar.gz` with all binaries, but should not we also provide OS specific downloads, specifically on GitHub? Maybe we can provide a generic tar.gz for Maven Central and upload OS specific tar.gz as assets to the GitHub Release?

Reply:
All fine. I just wanted to understand and clarify. What confused me is that in the same assembly declaration we have includes for all 3 OS types.

Creating downloads per OS is exactly what we had discussed and decided so far. So you did this perfectly correct.
I also see the advantage that this way we only have a single `ide[.exe]` binary. Therefore our bash wrapper script will just call that binary and life is easy. What if we would create one big release containing `ide` binary for linux and for mac x64 and mac aarm64? Would make all more complicated.
A little bit interesting will be the case that one is working on windows but also wants to use IDEasy inside WSL sharing the same installation (esp. `IDE_ROOT`). Maybe then such user can manually install the linux binary aside so he has a `bin` folder with both `ide.exe` and linux elf `ide` binary and we can make this work properly from the `ide` bash wrapper script. However, nothing to solve right now. Considered done.